suite/spdz/src/main/java/dk/alexandra/fresco/suite/spdz/datatypes/SpdzSInt.java

```diff
@@ -2,23 +2,130 @@
 
 import dk.alexandra.fresco.framework.value.SInt;
 import java.io.Serializable;
+import java.math.BigInteger;
 
 public class SpdzSInt implements SInt, Serializable {
 
-  private static final long serialVersionUID = -9048612329603301195L;
-  public SpdzElement value;
+  private static final long serialVersionUID = 8828769687281856043L;
+  private final BigInteger share;
+  private final BigInteger mac;
+  private final BigInteger mod;
 
-  public SpdzSInt() {
-    this.value = null;
+  /**
+   * Create a SpdzSInt containing a share, mac and modulus.
+   *
+   * @param share The share
+   * @param mac The mac
+   * @param modulus The modulus
+   */
+  public SpdzSInt(BigInteger share, BigInteger mac, BigInteger modulus) {
+    this.share = share;
+    this.mac = mac;
+    this.mod = modulus;
   }
 
-  public SpdzSInt(SpdzElement e) {
-    this.value = e;
+  //Communication methods
+
+  /**
+   * Create a SpdzSInt containing a share, mac and modulus. This constructor handles serialized
+   * data.
+   *
+   * @param data Array of bytes containing the share and mac. First half contains the share, second
+   * contains the mac
+   * @param modulus The modulus
+   * @param modulusSize The size of the share and mac
+   */
+  public SpdzSInt(byte[] data, BigInteger modulus, int modulusSize) {
+    byte[] shareBytes = new byte[modulusSize];
+    byte[] macBytes = new byte[modulusSize];
+    for (int i = 0; i < data.length / 2; i++) {
+      shareBytes[i] = data[i];
+      macBytes[i] = data[modulusSize + i];
+    }
+    this.share = new BigInteger(shareBytes);
+    this.mac = new BigInteger(macBytes);
+    mod = modulus;
+  }
+
+  //get operations
+  public BigInteger getShare() {
+    return share;
+  }
+
+  public BigInteger getMac() {
+    return mac;
+  }
+
+  /**
+   * Adds two SpdzSInt.
+   *
+   * @param e The element to add
+   * @return The sum
+   */
+  public SpdzSInt add(SpdzSInt e) {
+    BigInteger share = this.share.add(e.getShare()).mod(mod);
+    BigInteger mac = this.mac.add(e.getMac()).mod(mod);
+    return new SpdzSInt(share, mac, this.mod);
+  }
+
+  /**
+   * Add a public value to this element.
```

Comment:
element?

Reply:
This seems to be some problems with copy pasting the java doc from the old SpdzElement class.